File: ConversantSDKSampleAppObjC/ConversantSDKSampleAppObjC/ViewController.m
//
//  ViewController.m
//  ConversantSDKSampleAppObjC
//
//  Created by Daniel Kanaan on 4/14/17.
//  Copyright © 2017 Daniel Kanaan. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextView *feedbackTextView;
@property (weak, nonatomic) IBOutlet UIButton *bannerAdButton;

@end

@implementation ViewController


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (IBAction)loadAd:(id)sender {
    ConversantInterstitialAd *ad = [[ConversantInterstitialAd alloc] initWithDelegate: self];
    [ad fetch];
    [self appendFeedback: @"Fetching interstitial"];
}
- (IBAction) bannerAd: (UIButton*) sender {
    ConversantConfiguration.defaultConfiguration.autoDisplay = false;
    ConversantAdView * adView;
    //Check if the ad already exists
    adView = (ConversantAdView *)[self.view viewWithTag:9];
    if (adView != NULL) {
        if (adView.isReady) {
            [adView display];
            [sender setTitle:@"Fetch Again" forState:UIControlStateNormal];
            return;
        }
    } else {
        //If there is no ad, create an ad
        adView = [[ConversantAdView alloc] initWithAdFormat:ConversantAdFormatMobileBannerAd delegate: self];
        //Add it to the view heirarchy
        [self.view addSubview:adView];
        //Give it a tag so we can get it later
        [adView setTag:9];
    }
    //Set the frame of the ad to center below the button
    CGRect  newFrame = CGRectMake(sender.frame.origin.x + (sender.frame.size.width / 2) - (320 / 2), sender.frame.origin.y + sender.frame.size.height + 5, 320, 50 );
    [adView setFrame:newFrame];
    //fetch the ad
    [adView fetch];
    [self appendFeedback:@"Fetching banner"];
    //set the button to new text
    [sender setTitle:@"Display" forState:UIControlStateNormal];
}

//Delegate function for load failure.  Here we will present an Alert
- (void)interstitialAdLoadFailedWithAd:(ConversantInterstitialAd *)ad error:(ConversantError *)error {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Oops" message:[NSString stringWithFormat:@"Loading failed with error %@", error.description] preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction *ok = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [alert dismissViewControllerAnimated:true completion:NULL];
    }];
    [alert addAction:ok];
    [self presentViewController:alert animated:YES completion: nil];
}
//Delegate function for banner load failure.  Here we will present an Alert
- (void)bannerAdLoadFailedWithAd:(ConversantAdView *)ad error:(ConversantError *)error {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Oops" message:[NSString stringWithFormat:@"Loading failed with error %@", error.description] preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction *ok = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [alert dismissViewControllerAnimated:true completion:NULL];
    }];
    [alert addAction:ok];
    [self presentViewController:alert animated:YES completion: nil];
}

//Delegate function for successful load
- (void)interstitialAdLoadCompleteWithAd:(ConversantInterstitialAd *)ad {
    [ad presentFrom:self];
    [self appendFeedback:@"Interstitial Loaded"];
}

- (void)bannerAdLoadCompleteWithAd:(ConversantAdView *)ad {
    [self appendFeedback:@"Banner Loaded"];
    if (ad.isReady) {
        [ad display];
        [_bannerAdButton setTitle:@"Fetch again" forState: UIControlStateNormal];
    } else {
        [_bannerAdButton setTitle:@"Display" forState:UIControlStateNormal];
    }
   // [ad setFrame:CGRectMake(self.view.bounds.origin.x, self.view.bounds.origin.y + self.view.bounds.size.height - 50, ad.frame.size.width, ad.frame.size.height)];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

//Helper functions

- (void) appendFeedback: (NSString * ) text {
    [self.feedbackTextView setText:[[self.feedbackTextView.text stringByAppendingString: text] stringByAppendingString:@"\n"]];
    double bottom = self.feedbackTextView.contentSize.height - self.feedbackTextView.bounds.size.height;
    [self.feedbackTextView setContentOffset:CGPointMake(0, bottom) animated: YES];
}


//Other Delegate Functions
- (void)bannerAdWillPresentScreenWithAd:(ConversantAdView *)ad {
    [self appendFeedback:NSStringFromSelector(_cmd)];
}
- (void)bannerAdDidDismissScreenWithAd:(ConversantAdView *)ad {
    [self appendFeedback:NSStringFromSelector(_cmd)];
}
- (void)bannerAdWillDismissScreenWithAd:(ConversantAdView *)ad {
    [self appendFeedback:NSStringFromSelector(_cmd)];
}
- (void)interstitialAdWillAppearWithAd:(ConversantInterstitialAd *)ad {
    [self appendFeedback:NSStringFromSelector(_cmd)];
}
- (void)interstitialAdWillDisappearWithAd:(ConversantInterstitialAd *)ad {
    [self appendFeedback:NSStringFromSelector(_cmd)];
}
- (void)interstitialAdDidDisappearWithAd:(ConversantInterstitialAd *)ad {
    [self appendFeedback:NSStringFromSelector(_cmd)];
}



@end

File: ConversantSDKSampleAppSwift/ConversantSDKSampleAppSwift/ViewController.swift
//
//  ViewController.swift
//  ConversantSDKSampleAppSwift
//
//  Created by Daniel Kanaan on 4/14/17.
//  Copyright © 2017 Daniel Kanaan. All rights reserved.
//

import UIKit
import ConversantSDK

class ViewController: UIViewController, ConversantInterstitialAdDelegate, ConversantBannerAdDelegate {
    
    @IBOutlet weak var feedbackTextView: UITextView!
    @IBOutlet weak var bannerButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    @IBAction func loadAd () {
        //Create an Interstitial Ad
        let interstitialAd = ConversantInterstitialAd(delegate: self)
        //Fetch the interstitial Ad
        interstitialAd.fetch()
        appendFeedback(text: "Fetching interstitial")
    }
    @IBAction func bannerAd(_ sender: UIButton) {
        ConversantConfiguration.defaultConfiguration.autoDisplay = false
        var adView:ConversantAdView!
        //Check if the ad already exists
        if let currentAd = view.viewWithTag(9) as? ConversantAdView {
            adView = currentAd
            if adView.isReady {
                adView.display()
                sender.setTitle("Fetch again", for: .normal)
                return
            }
        } else {
            //If there is no ad, create an ad
            adView = ConversantAdView(adFormat: .mobileBannerAd, delegate: self)
            //Add it to the view heirarchy
            view.addSubview(adView)
            //Give it a tag so we can get it later
            adView.tag = 9
        }
        //Set the frame of the ad to center below the button
        let newFrame = CGRect(x:sender.frame.origin.x + (sender.frame.width / 2) - (ConversantAdFormat.mobileBannerAd.width / 2), y:sender.frame.origin.y + sender.frame.height + 5, width: ConversantAdFormat.mobileBannerAd.width, height: ConversantAdFormat.mobileBannerAd.height)
        adView.frame = newFrame
        //fetch the ad
        adView.fetch()
        appendFeedback(text: "Fetching banner")
        //set the button to new text
        sender.setTitle("Display", for: .normal)
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    //Delegate function for successful load
    func interstitialAdLoadComplete(ad: ConversantInterstitialAd) {
        //Present the interstitial ad from this view controller
        ad.present(from: self)
        appendFeedback(text: "Interstitial Loaded")
    }
    
    //Delegate function for load failure.  Here we will present an Alert
    func interstitialAdLoadFailed(ad: ConversantInterstitialAd, error: ConversantError) {
        //Create a message using the error.description property
        let alert = UIAlertController(title: "Failed", message: "Ad Load Failed with error: \(error.description)", preferredStyle: .alert)
        let ok = UIAlertAction(title: "OK", style: .default) { (_) in
            alert.dismiss(animated: true, completion: nil)
        }
        alert.addAction(ok)
        present(alert, animated: true, completion: nil)
    }
    
    func bannerAdLoadComplete(ad: ConversantAdView) {
        appendFeedback(text: "Banner Loaded")
        if ad.isReady {
            ad.display()
            bannerButton.setTitle("Fetch again", for: .normal)
        } else {
            bannerButton.setTitle("Display", for:.normal)
        }
    }
    
    //Delegate function for banner load failure.  Here we will present an Alert
    func bannerAdLoadFailed(ad: ConversantAdView, error: ConversantError) {
        //Create a message using the error.description property
        let alert = UIAlertController(title: "Failed", message: "Ad Load Failed with error: \(error.description)", preferredStyle: .alert)
        let ok = UIAlertAction(title: "OK", style: .default) { (_) in
            alert.dismiss(animated: true, completion: nil)
        }
        alert.addAction(ok)
        present(alert, animated: true, completion: nil)
    }
    
    //Other Delegate Functions
    func bannerAdWillPresentScreen(ad: ConversantAdView) {
        appendFeedback(text: "\(#function)")
    }
    
    func bannerAdDidDismissScreen(ad: ConversantAdView) {
        appendFeedback(text: "\(#function)")
    }
    func bannerAdWillDismissScreen(ad: ConversantAdView) {
        appendFeedback(text: "\(#function)")
    }
    func interstitialAdWillAppear(ad: ConversantInterstitialAd) {
        appendFeedback(text: "\(#function)")
    }
    func interstitialAdWillDisappear(ad: ConversantInterstitialAd) {
        appendFeedback(text: "\(#function)")
    }
    func interstitialAdDidDisappear(ad: ConversantInterstitialAd) {
        appendFeedback(text: "\(#function)")
    }
    //Helper functions
    
    func appendFeedback(text:String) {
        feedbackTextView.text = feedbackTextView.text +  text + "\n"
        let bottom = feedbackTextView.contentSize.height - feedbackTextView.bounds.size.height
        feedbackTextView.setContentOffset(CGPoint(x: 0, y: bottom), animated: true)
    }
}

